test/tests/transact.ts

```diff
@@ -192,55 +137,227 @@ suite('transact', function () {
             test('true', async function () {
+                const {action, session} = await mockData()
                 const result = await session.transact({action}, {broadcast: true})
                 assetValidTransactResponse(result)
             })
             test('false', async function () {
+                const {action, session} = await mockData()
                 const result = await session.transact({action}, {broadcast: false})
                 assetValidTransactResponse(result)
             })
         })
+        suite('transactPlugins', function () {
+            test('inherit', async function () {
+                const {action} = await mockData()
+                const session = new Session({
+                    ...mockSessionOptions,
+                    transactPlugins: [new MockTransactResourceProviderPlugin()],
+                })
+                const result = await session.transact({action})
+                assetValidTransactResponse(result)
+                if (result && result.transaction && result.transaction.actions) {
+                    assert.lengthOf(result.transaction.actions, 2)
+                } else {
+                    assert.fail('Transaction with actions was not returned in result.')
+                }
+            })
+            test('override', async function () {
+                const {action, session} = await mockData()
+                const result = await session.transact(
+                    {action},
+                    {
+                        transactPlugins: [new MockTransactResourceProviderPlugin()],
+                    }
+                )
+                assetValidTransactResponse(result)
+                if (result && result.transaction && result.transaction.actions) {
+                    assert.lengthOf(result.transaction.actions, 2)
+                } else {
+                    assert.fail('Transaction with actions was not returned in result.')
+                }
+            })
+            test('triggers', async function () {
+                const {action, session} = await mockData()
+                const result = await session.transact(
+                    {action},
+                    {transactPlugins: [new MockTransactPlugin()]}
+                )
+                assetValidTransactResponse(result)
+            })
+        })
+        suite('transactPluginsOptions', function () {
+            test('transact', async function () {
+                const {action} = await mockData()
+                const session = new Session({
+                    ...mockSessionOptions,
+                    transactPlugins: [new MockTransactResourceProviderPlugin()],
+                })
+                const result = await session.transact(
+                    {action},
+                    {
+                        transactPluginsOptions: {
+                            disableExamplePlugin: true,
+                        },
+                    }
+                )
```

Comment:
Options can be passed to plugins via the `transactPluginOptions` field during a `session.transact` call.

Reply:
I was hoping this would show more of the code, rather than just 4 lines 😅